web_app/db/crud.py

```diff
@@ -512,3 +512,36 @@ def delete_telegram_user(self, telegram_id: str) -> None:
         user = self.get_user_by_telegram_id(telegram_id)
         if user:
             self.delete_object(user, user.id)
+
+    def set_notification_allowed(self, telegram_id: str = None, wallet_id: str = None, is_allowed: bool = None) -> TelegramUser:
+        """
+        Toggles or sets is_allowed_notification for a TelegramUser, creating a new user if none exists.
+        Either telegram_id or wallet_id must be provided.
+        
+        :param telegram_id: str, optional
+        :param wallet_id: str, optional
+        :param is_allowed: bool, optional - if not provided, toggles the current value
+        :return: TelegramUser
+        """
+        if not telegram_id and not wallet_id:
+            raise ValueError("Either telegram_id or wallet_id must be provided")
+
+        with self.Session() as session:
+            user = None
+            if telegram_id:
```

Comment:
Please, simplify it, just check if you have instance `TelegramUser` first, if not, create it